crates/ott-balancer/src/balancer.rs

```diff
@@ -696,3 +696,87 @@ pub async fn dispatch_monolith_message(
 
     Ok(())
 }
+
+#[cfg(test)]
+mod test {
+    use std::net::Ipv4Addr;
+
+    use crate::discovery::{HostOrIp, MonolithConnectionConfig};
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_clients_add_remove() {
+        // a bunch of setup
+        let room_name = RoomName::from("test");
+        let mut ctx = BalancerContext::new();
+        let (monolith_outbound_tx, _monolith_outbound_rx) = tokio::sync::mpsc::channel(100);
+        let monolith_outbound_tx = Arc::new(monolith_outbound_tx);
+        let (client_inbound_tx, _client_inbound_rx) = tokio::sync::mpsc::channel(100);
+        let (client_unicast_tx, _client_unicast_rx) = tokio::sync::mpsc::channel(100);
+        let monolith_id = uuid::Uuid::new_v4().into();
+        let monolith = BalancerMonolith::new(
+            NewMonolith {
+                id: monolith_id,
+                region: "unknown".into(),
+                config: MonolithConnectionConfig {
+                    host: HostOrIp::Ip(Ipv4Addr::LOCALHOST.into()),
+                    port: 3002,
+                },
+                proxy_port: 3000,
+            },
+            monolith_outbound_tx,
```

Comment:
Why are these defined in a different scope to that of the other values? Is it because the `new()` constructor takes them as arguments?

Reply:
The channels need to be defined at the function level context so they stay alive for the entire unit test. Otherwise the channels get disposed immediately and the test will fail.